docs/dark.md

```diff
@@ -0,0 +1,143 @@
+> &{[(\exists \omega_1,\omega_2,\omega_3,\omega_4,\omega_5,\omega_6)(\exists_b \omega_7,\omega_8)]} &&&\\
+> &[(\omega_2 = \omega_1\mu_1 −\mu^2_1 −1)\land(\omega_2 −\mu_3 −1 \geq 0)\land &&&(E1-E2) \\
+> &(\mu_3 −(\mu_1 −\omega_1)\omega_7 −\omega_8 = \omega_2\omega_3))\land(\omega_1 −2 \geq 0)\land &&&(E3-E4)\\
+> &((\omega_1 −2)^2 −(\mu_1 +2)(\omega_1 −2)\omega_5 −\omega^2_5 =1)\land &&&(E5) \\
+> &(\omega_1 −2=\mu_2+\omega_6(\mu_1 +2))\land(\omega_7 \geq 0)\land(\omega_7 <\omega_8)\land &&&(E6-E8)\\
+> &(\omega^2_7 − \omega_1\omega_7\omega_8 −\omega^2_8 =1)\land(\omega_7 = \mu_2 +\omega_4(\omega_{1} −2)],&&&(E9-E10)
+> \end{aligned}
+> $$
+>
+> where $\exists_b$ signifies a bounded quantifier in the following sense: if $\mu_3 =\mu_1^{\mu_2}$ then $E(\mu_1,\mu_2,\mu_3)$ is
+> true with $W = Θ(\mu_2^2log\mu_1)=o(M^2).$ On the other hand, if $\mu_{3 \neq}\mu_1^{\mu_2}$ then either $E(\mu_1,\mu_2,\mu_3)$ is
+> false, or it is true but the intermediate witnesses $\omega_7$ and $\omega_8$ have length $\Omega(\mu_3log\mu_3)$, which is equal to
+> $\Omega(2^M · M)$ in the worst case.
+
+Basically, if $\mu_3 = \mu_1^{\mu_2}$, we can
+efficiently find our witness for $R_S$ that has sub-quadratic length. If $\mu_3 \neq \mu_1^{\mu_2}$, either the witness for $R_S$ does
+not exist, or the witness has a large length, $\Omega(\mu_3log\mu_3)$ - which makes it impractical to use as a fake "proof" that $\mu_3
+= \mu_1^{\mu_2}$. We can also take care of edge cases like $\mu_1=0,1, \mu_2=0,1,2,\mu_3=0$ easily with our "fact" as well. This shows
+that we can completely prove $\mu_3 = \mu_1^{\mu_2}$ with DARK, zero-knowledge. The auxiliary witnesses can be calculated by below
+formulas:
+
+- $\omega_{1} \leftarrow (\mu_1 +2)^{[\mu_2+1]} +2$
+- $\omega_2 \leftarrow \omega_1\mu1 − \mu_{1}^{2}-1$
+- $\omega_{3}\leftarrow(\mu_{3}-(\mu_{1}-\omega_{1})\omega_{1}^{[\mu_{2}]}-\omega_{1}^{[\mu_{2}+1]})/\omega_{2}$
+- $\omega_{4}\leftarrow(\omega_{8}-\mu_{2})/(\omega_{1}-2)$
+- $\omega_{5}\leftarrow(\mu_{1}+2)^{[\mu_{2}+2]}$
+- $\omega_{6}\leftarrow(\omega_{1}-2-\mu_{2})/\dot{(\mu_{1}+2)}$
+- $\omega_{7}\leftarrow\omega_{1}^{[\mu_{2}]}$
+- $\omega_{8}\leftarrow\omega_{1}^{[\mu_{2}+1]}$
+
+### Bounded Arithmetic
+<a id="bounded-arithmetic"></a>
+Let's denote some new operations:
+
+- $\sigma(x)=x+1$
+- $x \overset{.}{-} y := max(x − y,0)$
+- $|x| := \lfloor {log_2(x) +1} \rfloor$
+- $MSP(x,i) := \lfloor x/2^i\rfloor$
+- $x \sharp y := 2^{|x|·|y|}$
+
+We have following representations:
+
+- $\sigma: [\mu_2 = \sigma(\mu_1)] \equiv [\mu_2 − \mu_1 − 1]$
+- $\leq$: proves $b−a \geq 0$ to show $a \leq b$.
+- $\overset{.}{-}$: $[\mu_3 = \mu_1 \overset{.}{-} \mu_2] \equiv [((\mu_1 − \mu_2 = \mu_3) \land (\mu_1 \geq \mu_2)) \vee (\mu_3
+  =0\land\mu_1 <\mu_2)]$
+- $\lfloor x/2 \rfloor: [\mu_2 = \lfloor \mu_1/2 \rfloor ] \equiv [(\mu_1 = 2\mu_2) \vee (\mu_1 =2\mu_2 + 1)]$
+- $|x|: [\mu_2 = |\mu_1|] \equiv [\omega_1 =2^{\mu_2} \land\omega_1 \leq  2(\mu_1 +1)\land(\mu_1 +1)<\omega_1]$
```

Comment:
In case $\mu_1 = 3$ =>  $\mu_2 = 2$ 
=> $\omega_1 = 4$ and $(\mu_1 +1) = 4$
So it should be  $(\mu_1 +1) \leq \omega_1$, right?